scio-core/src/main/scala/com/spotify/scio/io/Tap.scala

```diff
@@ -37,6 +36,9 @@ import scala.reflect.ClassTag
  */
 trait Tap[T] { self =>
 
+  /** Parent Taps this Tap's data came from after [[Tap.map]] */
+  val parent: Seq[Tap[_]] = Nil
```

Comment:
You don't really need a `Seq` right? Just `Tap` should be sufficient since its parent also has a `parent` field. Like a linked list.